gammapy/maps/geom.py

```diff
@@ -974,6 +1162,25 @@ def to_cube(self, axes):
         """
         pass
 
+    def coord_to_tuple(self, coord):
+        """Generate a coordinate tuple compatible with this geometry.
+
+        Parameters
+        ----------
+        coord : `~MapCoord`
+        """
+        if self.ndim != coord.ndim:
+            raise ValueError
+
+        if not coord._match_by_name:
```

Comment:
@woodmd - PyCharm complains here that you're accessing the private `MapCoord._match_by_name` from outside the class (here from `MapGeom.coord_to_tuple`). Seems like a valid complaint? What would you suggest here?

Reply:
Yes this was just laziness on my part.  I'll make `match_by_name` into a public attribute.